test/Java10andUp/src/org/openj9/test/java/lang/management/ThreadMXBean/APITestJava10.java

```diff
@@ -0,0 +1,252 @@
+				m.invoke(this, Integer.valueOf(stackDepth));
+			} catch (NoSuchMethodException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+				fail("Unexpected exception", e);
+			}
+		}
+	}
+
+	class LambdaTestThread extends TestThread {
+		public LambdaTestThread(int depth) {
+			super(depth);
+		}
+
+		@Override
+		public void run() {
+			try {
+				UnaryOperator<Integer> myLambda = s -> myMethod(s);
+				myLambda.apply(Integer.valueOf(stackDepth));
+			} catch (IllegalArgumentException e) {
+				fail("Unexpected exception", e);
+			}
+		}
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testDumpLimits() throws InterruptedException {
+		logger.debug("testDumpLimits");
+		ArrayList<Thread> myThreads = new ArrayList<>();
+		Thread currentThread = Thread.currentThread();
+		long myId = currentThread.getId();
+		myThreads.add(currentThread);
+		long ids[] = new long[MAXDEPTH];
+		HashMap<Long, Integer> expectedDepths = new HashMap<>();
+		for (int requestedDepth = 0; requestedDepth < MAXDEPTH; requestedDepth++) {
+			Thread t = new TestThread(requestedDepth);
+			myThreads.add(t);
+			long id = t.getId();
+			logger.debug("start "+id);
+			t.start();
+			while (t.getState() != Thread.State.WAITING) {
+				try {
+					Thread.sleep(1);
+				} catch (InterruptedException e) {
+					// ignore
+				}
+			}
+			ids[requestedDepth] = id;
+			/*
+			 * every thread will have one stack frame for the run() method,
+			 * plus the specified number of extra frames.
+			 */
+			expectedDepths.put(id, requestedDepth + 1);
+		}
+
+		for (int maxDepth = 0; maxDepth < MAXDEPTH; maxDepth++) {
+			logger.debug("maxDepth = "+maxDepth);
+			for (ThreadInfo ti: myBean.dumpAllThreads(false, false, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				Integer requestedDepthTemp = expectedDepths.get(id);
+				if (Objects.isNull(requestedDepthTemp)) {
+					/* non-test thread */
+					continue;
+				}
+				int requestedDepth = requestedDepthTemp;
+				if (id != myId) {
+					checkDepth(requestedDepth, maxDepth, trace, "dumpAllThreads");
+				} else {
+					Assert.assertTrue(trace.length <= maxDepth, "dumpAllThreads: Wrong stack size when maxDepth = "+maxDepth);
+				}
+			}
+			for (ThreadInfo ti: myBean.getThreadInfo(ids, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				Integer requestedDepthTemp = expectedDepths.get(id);
+				if (null == requestedDepthTemp) {
+					/* non-test thread */
+					continue;
+				}
+				int requestedDepth = requestedDepthTemp;
+				if (id != myId) {
+					checkDepth(requestedDepth, maxDepth, trace, "getThreadInfo");
+				} else {
+					Assert.assertTrue(trace.length <= maxDepth, "getThreadInfo: Wrong stack size when maxDepth = "+maxDepth);
+				}
+			}
+		}
+		synchronized (waitObject) {
+			waitObject.notifyAll();
+		}
+		for (Thread t: myThreads) {
+			if (Thread.currentThread().equals(t)) {
```

Comment:
There doesn't seem to be any point in added the current thread to `myThreads`.

Reply:
True.  Fixed.